mosaicml/llm-foundry
# Torch2 (#177)

move https://github.com/mosaicml/llm-foundry/pull/149 to main repo (from a fork)



uses https://github.com/mosaicml/llm-foundry/pull/147 as a springboard to updt torch

In interactive instance, I installed torch2 req and everything works fine

125M models was getting good (the same) MFU from the same exact config in both torch1.13 and torch2

Note: torch2 version `pip list` has both triton version:
```
torch                  2.0.1+cu118

triton                 2.0.0
triton-pre-mlir        2.0.0
```
doesn't seem to matter

Note: this does not use `torch.compile()` (but there is no reason it shouldn't)

Note: `flash-attn` is still installed. `xentropy-cuda-lib` is also still installed; I'm not setting `loss_fn` so mpt defaults to using `fused_crossentropy` for both settings.

Biggest low probability risk: this old version of triton does not compile / work for H100s... 👀 
Risk: `triton_pre_mlir` has no support and will never be updated.

~~Still need to test at scale / convergence~~
see torch2 vs torch1.13 produce the same results [here](https://wandb.ai/mosaic-ml/torch2_test)

cc @sashaDoubov (enables torch2 for muP dev)
cc @dskhudia enables torch2 and torch.compile() with triton attn impl 


old pr commits:

* make triton attn req mlri tagged triton

* add comment

* updt err

* clean up req / install

* exclude HazyR flash attn from pyright

* lint

* exclude flash_attn_triton.py from pyright

* updt torch version & install instructions

* add extra install instructions for installing CMake

* lint

* adding torch1.13 and torch2 testing matrix
